### DISCUSSION TITLE
Finding material of an element starting from its GlobalID

### DISCUSSION THREAD

**Discussion by LouiseToye** [IfcOpenShell/IfcOpenShell#2834] [Q&A]
Hi all!

I would like to find a way to match the right material to a list of elements using the IFC file of a project. The elements are not complex, i only worked with slabs, columns and beams.

Using ifcopenshell.util.element.get_material i can find the IfcMaterialProfileSet for each element, but how do i get from this set to the IfcMaterial?

**Comment by Moult** [comment]
```[mp.Material for mp in profile_set.MaterialProfiles]```
